### PR TITLE
style: Enable errcheck linter and associated recommendations

### PR DESCRIPTION
**Pull request by bnevis-i**
See #3565

## PR Checklist
Please check if your PR fulfills the following requirements:

- [X] I am not introducing a breaking change (if you are, flag in conventional commit message with `BREAKING CHANGE:` describing the break)
- [X] I am not introducing a new dependency (add notes below if you are)
- [ ] I have added unit tests for the new feature or bug fix (if not, why?)  Not adding functionality.
- [ ] I have fully tested (add details below) this the new feature or bug fix (if not, why?)  Not adding functionality.
- [ ] I have opened a PR for the related docs change (if not, why?) Not applicable.

## Testing Instructions
make test; make docker;  make run dev in edgex-compose

### CODE REVIEW

File: internal/support/notifications/application/distributioncoordinator.go [hunk 1]
```diff
@@ -36,7 +36,7 @@ func distribute(dic *di.Container, n models.Notification) errors.EdgeX {
 		}
 		for _, address := range sub.Channels {
 			// Async transmit the notification to improve the performance
-			go transmit(dic, n, sub, address)
+			go transmit(dic, n, sub, address) // nolint:errcheck
```
Comment (bnevis-i) [inline internal/support/notifications/application/distributioncoordinator.go:39]: Seems clearer to just suppress the error.

File: internal/support/notifications/application/notification.go [hunk 2]
```diff
@@ -35,7 +35,7 @@ func AddNotification(n models.Notification, ctx context.Context, dic *di.Contain
 		addedNotification.Id,
 		correlation.FromContext(ctx))
 
-	go distribute(dic, addedNotification)
+	go distribute(dic, addedNotification) // nolint:errcheck
```
Comment (bnevis-i) [inline internal/support/notifications/application/notification.go:38]: Seems clearer to just suppress the error.

File: internal/support/notifications/application/send.go [hunk 3]
```diff
@@ -112,7 +112,7 @@ func escalatedSend(dic *di.Container, n models.Notification, trans models.Transm
 	}
 
 	for _, address := range sub.Channels {
-		go transmit(dic, escalated, sub, address)
+		go transmit(dic, escalated, sub, address) // nolint:errcheck
```
Comment (bnevis-i) [inline internal/support/notifications/application/send.go:115]: Seems clearer to just suppress the error.

**Review by cloudxxx8 — APPROVED** [review]
LGTM

**Comment by cloudxxx8** [comment]
recheck

**Pull request merged**